modules/sw360/src/test/java/org/eclipse/sw360/antenna/workflow/generator/SW360UpdaterTest.java

```diff
@@ -129,4 +127,47 @@ public String getLinkStr() {
         return artifact;
     }
 
+    @Test
+    public void testMainLicenseIsEmptyAndFinalLicenseNull() {
+        Artifact artifact  = new Artifact("JSON");
+        artifact.setProprietary(false);
+        artifact.addFact(new MavenCoordinates("artifactId(test)", "org.group.id", "1.2.3"));
+        artifact.addFact(new DeclaredLicenseInformation(new LicenseInformation() {
+            @Override
+            public String evaluate() {
+                return "";
+            }
+
+            @Override
+            public String evaluateLong() {
+                return null;
+            }
+
+            @Override
+            public boolean isEmpty() {
+                return false;
+            }
+
+            @Override
+            public List<License> getLicenses() {
+                License license = new License();
+                license.setName("");
+                return Collections.singletonList(license);
+            }
+
+            @Override
+            public String getLinkStr() {
+                return null;
+            }
+        }));
+
+        SW360Release release = new SW360Release();
+        SW360Component sw360Component = new SW360Component();
+        sw360Component.setName("artifactId(test)");
+        SW360ReleaseAdapterUtils.prepareRelease(release, sw360Component, Collections.EMPTY_SET, artifact);
+
+        assertThat(release.getFinalLicense() == null).isTrue();
```

Comment:
If the `FinalLicense` is nullable, it might be better to use an `Optional` to make that explicit

Reply:
I think there should be
```suggestion
        assertThat(release.getFinalLicense()).isNull();
```